internal/metricdata/subscriber.go

```diff
@@ -0,0 +1,215 @@
+// SPDX-FileCopyrightText: 2022-present Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package metricdata
+
+import (
+	"fmt"
+	"sync/atomic"
+
+	"github.com/omec-project/metricfunc/internal/promclient"
+	"github.com/omec-project/metricfunc/logger"
+	"github.com/omec-project/metricfunc/pkg/metricinfo"
+)
+
+var (
+	smContextCount  uint64
+	smContextActive uint64
+)
+
+func incSMContextActive() uint64 {
+	atomic.AddUint64(&smContextActive, 1)
+	return smContextActive
+}
+
+func decSMContextActive() uint64 {
+	atomic.AddUint64(&smContextActive, ^uint64(0))
+	return smContextActive
+}
+
+func GetSMContextCount() uint64 {
+	atomic.AddUint64(&smContextCount, 1)
+	return smContextCount
+}
+
+func HandleSubscriberEvent(subsData *metricinfo.CoreSubscriberData, sourceNf metricinfo.NfType) {
+	switch subsData.Operation {
+	case metricinfo.SubsOpAdd:
+		storeSubscriber(&subsData.Subscriber, sourceNf)
+	case metricinfo.SubsOpMod:
+		updateSubscriber(&subsData.Subscriber, sourceNf)
+	case metricinfo.SubsOpDel:
+		deleteSubscriber(&subsData.Subscriber, sourceNf)
+	default:
+		logger.CacheLog.Errorf("unknown smf subsriber operation [%v] ", subsData.Operation)
+	}
+}
+
+func storeSubscriber(sub *metricinfo.CoreSubscriber, sourceNf metricinfo.NfType) error {
```

Comment:
Name seems mis leading. Initially I thought its storing in DB. But looks like its updating/creating local in memory subscriber.

Reply:
DB save operation shall be extended in same function